pygmt/io/gmtread.py

```diff
@@ -0,0 +1,120 @@
+"""
+Read a file into an appropriate object.
+"""
+
+from collections.abc import Mapping, Sequence
+from pathlib import PurePath
+from typing import Any, Literal
+
+import pandas as pd
+import xarray as xr
+from pygmt.clib import Session
+from pygmt.helpers import build_arg_list, is_nonstr_iter
+from pygmt.src.which import which
+
+
+def gmtread(
+    file: str | PurePath,
+    kind: Literal["dataset", "grid", "image"],
+    region: Sequence[float] | str | None = None,
+    header: int | None = None,
+    column_names: pd.Index | None = None,
+    dtype: type | Mapping[Any, type] | None = None,
+    index_col: str | int | None = None,
+) -> pd.DataFrame | xr.DataArray:
```

Comment:
On second thought, I'm thinking if we should make `gmtread` a private function for internal use only for now, the fact that it can read either tabular or grid/image files seems like a lot of magic.

Reply:
It seems the `gmtread` function is no longer needed if PR #3919 is implemented, right?

Reply:
Yes, not needed for grids/images, but we could still use `gmtread` for tabular datasets? Though let's think about https://github.com/GenericMappingTools/pygmt/pull/3673#discussion_r2048080261.